pallets/ajuna-awesome-avatars/src/tests.rs

```diff
@@ -14,11 +14,238 @@
+		new_test_ext().execute_with(|| {
+			let new_season = Season::new(1, 5, 10, 0, 0);
+
+			assert_ok!(AAA::new_season(Origin::signed(ALICE), new_season));
+			assert_eq!(
+				last_event(),
+				mock::Event::AAA(crate::Event::NewSeasonCreated(Season::new(1, 5, 10, 0, 0)))
+			);
+		});
+	}
+
+	#[test]
+	fn new_season_should_return_error_when_start_block_smaller_than_end_block_in_current_season() {
+		new_test_ext().execute_with(|| {
+			let new_season = Season::new(1, 5, 10, 0, 0);
+
+			assert_ok!(AAA::new_season(Origin::signed(ALICE), new_season));
+			assert_eq!(
+				last_event(),
+				mock::Event::AAA(crate::Event::NewSeasonCreated(Season::new(1, 5, 10, 0, 0)))
+			);
+
+			// ensure new season’s early access start > last season’s end
+			let new_season = Season::new(3, 7, 10, 0, 0);
+			assert_noop!(
+				AAA::new_season(Origin::signed(ALICE), new_season),
+				Error::<Test>::EarlyAccessStartsTooEarly
+			);
+		});
+	}
+
+	#[test]
+	fn new_season_should_return_error_when_early_access_block_greater_than_start() {
+		new_test_ext().execute_with(|| {
+			// ensure new season’s early access start < new season’s start
+			let new_season = Season::new(6, 3, 10, 0, 0);
+			assert_noop!(
+				AAA::new_season(Origin::signed(ALICE), new_season),
+				Error::<Test>::EarlyAccessStartsTooLate
+			);
+		});
+	}
+
+	#[test]
+	fn new_season_should_return_error_when_start_block_greater_than_end() {
+		new_test_ext().execute_with(|| {
+			// ensure new season’s start < new season’s end
+			let new_season = Season::new(11, 12, 10, 0, 0);
+			assert_noop!(
+				AAA::new_season(Origin::signed(ALICE), new_season),
+				Error::<Test>::SeasonStartsTooLate
+			);
+		});
+	}
+}
+
+pub mod update_season {
+	use frame_support::{assert_noop, assert_ok};
+
+	use crate::{ajuna_awesome_avatar::Season, mock::*, tests::ALICE, *};
+
+	#[test]
+	fn update_season_should_return_error_when_season_not_found() {
+		new_test_ext().execute_with(|| {
+			assert_noop!(
+				AAA::update_season(Origin::signed(ALICE), 10, Season::new(1, 12, 30, 1, 1)),
+				Error::<Test>::UnknownSeason
+			);
+		});
+	}
+
+	#[test]
+	fn update_season_should_return_error_when_season_to_update_ends_after_next_season_start() {
+		new_test_ext().execute_with(|| {
+			// Create two seasons
+			let first_season = Season::new(1, 5, 10, 0, 0);
+			assert_ok!(AAA::new_season(Origin::signed(ALICE), first_season.clone()));
+			assert_eq!(
+				last_event(),
+				mock::Event::AAA(crate::Event::NewSeasonCreated(first_season))
+			);
+			let second_season = Season::new(11, 15, 20, 0, 0);
+			assert_ok!(AAA::new_season(Origin::signed(ALICE), second_season.clone()));
+			assert_eq!(
+				last_event(),
+				mock::Event::AAA(crate::Event::NewSeasonCreated(second_season))
+			);
+
+			// Update the first one to end after the second has started
+			let first_season_update = Season::new(1, 5, 14, 0, 0);
+			assert_noop!(
+				AAA::update_season(Origin::signed(ALICE), 0, first_season_update),
```

Comment:
Don't forget to bump `season_id` by 1
```suggestion
				AAA::update_season(Origin::signed(ALICE), 1, first_season_update),
```

Reply:
Sorry I don't undestand this, I think it's ok. First season has index 0 and second is 1. It wants to update the season with index 0 and check that if the end is greater than the start of the following season (1) it returns error.

Reply:
I wanted to start season from `1` to avoid `season_id > 0` checks which just adds unnecessary complexity